### ISSUE TITLE
Transducer Loss using Triton

### ISSUE THREAD

**Issue by Adel-Moumen** [speechbrain/speechbrain#2736]
### Describe the bug

Okay. I think Triton is starting to be mature (e.g. Triton is now powering a large part of the `torch.compile` backend), and is now coming automatically with Torch (e.g. when doing pip install). This toolkit allows one to write CUDA kernel and leave the possibility to a better integration of AMD GPUs support for pytorch-based software. 

I would like to start a discussion on maybe, at least thinking of switching from our Numba Transducer loss to a Triton one. This would comes with many benefits: 
1. no Numba dependency anymore (always great to minimise dependency). 
2. in my experience, Triton >> Numba in terms of speed.
3. better support of hardware such as AMD, and I bet that Intel etc will also rely on Triton to integrate their GPUs with Torch. 

We could also leverage this opportunity to consider supporting other kernels such as cross-entropy, RoPE embeddings etc that are known to be faster while using Triton than Torch. 

Feel free to comment bellow. 

### Expected behaviour

bl

**Comment by Adel-Moumen** [comment]
CC: @asumagic @TParcollet  for discussion

**Comment by asumagic** [comment]
Could also consider supporting the feature set of k2's [`fast_rnnt`](https://github.com/k2-fsa/fast_rnnt) (pruned RNN-T, iirc something similar to FastEmit, etc.), but AFAIK that's basically strictly additional to it
